centos-7.template

```diff
@@ -111,50 +111,13 @@ rm -f cert-gen.base64
 chmod +x minishift-cert-gen
 mv minishift-cert-gen /usr/local/bin/
 
-# This unit file will take precedence over unit file which present /usr location
-# and it have daemon running using cert so when restart happen then also docker
-# daemon works as expected.
-cat << EOF > /etc/systemd/system/docker.service
-[Unit]
-Description=Docker Application Container Engine
-Documentation=http://docs.docker.com
-After=network.target rc-local.service
-Requires=docker-cleanup.timer
-
-[Service]
-Type=notify
-NotifyAccess=all
-Environment=GOTRACEBACK=crash
-Environment=DOCKER_HTTP_HOST_COMPAT=1
-Environment=PATH=/usr/libexec/docker:/usr/bin:/usr/sbin
-ExecStartPre=/usr/local/bin/minishift-cert-gen
-ExecStart=/usr/bin/dockerd-current -H tcp://0.0.0.0:2376 \
-                       -H unix:///var/run/docker.sock \
-                       --selinux-enabled \
-                       --log-driver=journald \
-                       --signature-verification=false \
-                       --add-runtime docker-runc=/usr/libexec/docker/docker-runc-current \
-                       --default-runtime=docker-runc \
-                       --exec-opt native.cgroupdriver=systemd \
-                       --userland-proxy-path=/usr/libexec/docker/docker-proxy-current \
-                       --storage-driver devicemapper \
-                       --tlsverify --tlscacert /etc/docker/ca.pem \
-                       --tlscert /etc/docker/server.pem \
-                       --tlskey /etc/docker/server-key.pem
-ExecReload=/bin/kill -s HUP $MAINPID
-LimitNOFILE=1048576
-LimitNPROC=1048576
-LimitCORE=infinity
-TimeoutStartSec=0
-Delegate=yes
-KillMode=process
-MountFlags=slave
-Environment=
-
-[Install]
-WantedBy=multi-user.target
-EOF
+# update docker.service file to exec the certificate generation script
+sed -i.back 's/ExecStart=/ExecStartPre=\/usr\/local\/bin\/minishift-cert-gen\n&/' /usr/lib/systemd/system/docker.service
```

Comment:
@praveenkumar Just curious to know what is the use of `&` at end of regex patter here.

Reply:
@budhrg It add null terminator https://stackoverflow.com/questions/32750591/sed-behavior-with-ampersand

Reply:
Thanks